### PR TITLE
feat: Add download report handler

### PR DESCRIPTION
**Pull request by FirestarJes**
# Description

Add download httptrigger and handler for settlement report

## Pull-request quality

- [x ] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [x] Tests are written and executed locally
- [ ] Subsystem tests have been tested (by manually deploying to `dev_002`)

### CODE REVIEW

File: source/dotnet/wholesale-api/Orchestrations/Functions/SettlementReports/SettlementReportDownloadTrigger.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+using Energinet.DataHub.Wholesale.Orchestrations.Functions.SettlementReports.Model;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.DurableTask.Client;
+
+namespace Energinet.DataHub.Wholesale.Orchestrations.Functions.SettlementReports;
+
+internal sealed class SettlementReportDownloadTrigger
+{
+    private readonly IUserContext<FrontendUser> _userContext;
```
Comment (michaelmajgaard) [inline source/dotnet/wholesale-api/Orchestrations/Functions/SettlementReports/SettlementReportDownloadTrigger.cs:30]: `_userContext` is not used

File: source/dotnet/wholesale-api/Orchestrations/Functions/SettlementReports/SettlementReportDownloadTrigger.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Net;
+using Azure;
+using Energinet.DataHub.Core.App.Common.Abstractions.Users;
+using Energinet.DataHub.Wholesale.CalculationResults.Interfaces.SettlementReports_v2;
+using Energinet.DataHub.Wholesale.CalculationResults.Interfaces.SettlementReports_v2.Models;
+using Energinet.DataHub.Wholesale.Common.Infrastructure.Security;
+using Energinet.DataHub.Wholesale.Orchestrations.Functions.SettlementReports.Model;
```
Comment (michaelmajgaard) [inline source/dotnet/wholesale-api/Orchestrations/Functions/SettlementReports/SettlementReportDownloadTrigger.cs:21]: unused usings

File: source/dotnet/wholesale-api/Orchestrations.IntegrationTests/Functions/SettlementReports/SettlementReportOrchestrationTests.cs [hunk 3]
```diff
@@ -118,6 +118,70 @@ public async Task MockExternalDependencies_WhenCallingDurableFunctionEndpoint_Or
...
+        using var request = new HttpRequestMessage(HttpMethod.Post, "api/RequestSettlementReport");
+        request.Headers.Add("Authorization", $"Bearer {CreateFakeInternalToken()}");
+        request.Content = new StringContent(
+            JsonConvert.SerializeObject(settlementReportRequest),
+            Encoding.UTF8,
+            "application/json");
+
+        var actualResponse = await Fixture.AppHostManager.HttpClient.SendAsync(request);
+        var httpResponse = await actualResponse.Content.ReadFromJsonAsync<SettlementReportHttpResponse>();
+        var completeOrchestrationStatus = await Fixture.DurableClient.WaitForInstanceCompletedAsync(
```
Comment (michaelmajgaard) [inline source/dotnet/wholesale-api/Orchestrations.IntegrationTests/Functions/SettlementReports/SettlementReportOrchestrationTests.cs:160]: `completeOrchestrationStatus` is not checked, should it have been?

File: source/dotnet/wholesale-api/Orchestrations/Functions/SettlementReports/SettlementReportDownloadTrigger.cs (hunk 1 above)
Comment (FirestarJes) [reply source/dotnet/wholesale-api/Orchestrations/Functions/SettlementReports/SettlementReportDownloadTrigger.cs:30]: It is now :)

File: source/dotnet/wholesale-api/Orchestrations/Functions/SettlementReports/SettlementReportDownloadTrigger.cs (hunk 2 above)
Comment (FirestarJes) [reply source/dotnet/wholesale-api/Orchestrations/Functions/SettlementReports/SettlementReportDownloadTrigger.cs:21]: fixed

File: source/dotnet/wholesale-api/Orchestrations.IntegrationTests/Functions/SettlementReports/SettlementReportOrchestrationTests.cs (hunk 3 above)
Comment (FirestarJes) [reply source/dotnet/wholesale-api/Orchestrations.IntegrationTests/Functions/SettlementReports/SettlementReportOrchestrationTests.cs:160]: No, it isn't what is checked here, so have been removed as a variable, we only need to wait for it to finish.

**Review by michaelmajgaard — APPROVED** [review]

**Pull request merged**